prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh

```diff
@@ -0,0 +1,176 @@
+    GARDENER_REGION
+    GARDENER_ZONES
+    GARDENER_CLUSTER_VERSION
+    GARDENER_KYMA_PROW_KUBECONFIG
+    GARDENER_KYMA_PROW_PROJECT_NAME
+    GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME
+)
+
+utils::check_required_vars "${requiredVars[@]}"
+
+if [[ $GARDENER_PROVIDER == "azure" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/azure.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/azure.sh"
+elif [[ $GARDENER_PROVIDER == "aws" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/aws.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/aws.sh"
+elif [[ $GARDENER_PROVIDER == "gcp" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/gcp.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    ## TODO what should I put here? Is this a backend?
+    log::error "GARDENER_PROVIDER ${GARDENER_PROVIDER} is not yet supported"
+    exit 1
+fi
+
+# nice cleanup on exit, be it succesful or on fail
+trap gardener::cleanup EXIT INT
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+export ERROR_LOGGING_GUARD
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set KYMA_SOURCE used by gardener::install_kyma
+# at the time of writing this comment, kyma-integration-gardener never sets BUILD_TYPE to "release"
+if [[ -n ${PULL_NUMBER} ]]; then
+    # In case of PR, operate on PR number
+    KYMA_SOURCE="PR-${PULL_NUMBER}"
+    export KYMA_SOURCE
+    # TODO maybe can be replaced with PULL_BASE_REF?
+elif [[ "${BUILD_TYPE}" == "release" ]]; then
+    readonly RELEASE_VERSION=$(cat "VERSION")
+    log::info "Reading release version from RELEASE_VERSION file, got: ${RELEASE_VERSION}"
+    KYMA_SOURCE="${RELEASE_VERSION}"
+    export KYMA_SOURCE
+else
+    # Otherwise (master), operate on triggering commit id
+    if [[ -n ${PULL_BASE_SHA} ]]; then
+        readonly COMMIT_ID="${PULL_BASE_SHA::8}"
+        KYMA_SOURCE="${COMMIT_ID}"
+        export KYMA_SOURCE
+    else
+        # periodic job, so default to main
+        KYMA_SOURCE="main"
+        export KYMA_SOURCE
+    fi
+fi
+
+# checks required vars and initializes gcloud/docker if necessary
+gardener::init
+
+# if MACHINE_TYPE is not set then use default one
+gardener::set_machine_type
+
+kyma::install_cli
+
+# currently only Azure generates overrides, but this may change in the future
+gardener::generate_overrides
+
+gardener::provision_cluster
+
+host::prepare_components_file(){
+cat << EOF > "$PWD/components.yaml"
+defaultNamespace: kyma-system
+prerequisites:
+  - name: "cluster-essentials"
+  - name: "istio"
+    namespace: "istio-system"
+  - name: "certificates"
+    namespace: "istio-system"
+components:
+  - name: "dex"
```

Comment:
What is the purpose of using Dex in `kyma alpha deploy` scenario?

Reply:
Current tests are dependent on Dex, admin-user is used in the tests.